components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2281,7 +2274,16 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         await this.guardCostCenterAccess(ctx, user.id, attrId, "update");
 
         const response = await this.usageService.getCostCenter({ attributionId });
-        if (response?.costCenter?.billingStrategy !== CostCenter_BillingStrategy.BILLING_STRATEGY_STRIPE) {
+
+        // backward compatibility for cost centers that were created before introduction of BillingStrategy
+        if (response.costCenter) {
+            const stripeSubscritpionId = await this.findStripeSubscriptionId(ctx, attributionId);
+            if (!stripeSubscritpionId) {
+                response.costCenter.billingStrategy = CostCenter_BillingStrategy.BILLING_STRATEGY_STRIPE;
+            }
+        }
+
+        if (response.costCenter?.billingStrategy !== CostCenter_BillingStrategy.BILLING_STRATEGY_STRIPE) {
```

Comment:
What will happen if I call this API directly from the browser without going through the Stripe setup flow? I believe this will cause my billing strategy to switch to Stripe, without a corresponding stripe subscription id

Reply:
It only set you the stripe billing strategy Gitpod finds a stipe subscription.

Reply:
Oops! the guard is wrong. Will update.